packages/core/lib/debug/printer.js

```diff
@@ -209,10 +210,33 @@ class DebugPrinter {
       this.config.logger.log(DebugUtils.formatStack(step.stack));
       this.config.logger.log("");
     }
-    this.config.logger.log(
-      DebugUtils.formatInstruction(traceIndex + 1, totalSteps, instruction)
-    );
-    this.config.logger.log(DebugUtils.formatPC(step.pc));
+
+    // printout instructions
+    const previousInstructions = 3;
+    const upcomingInstructions = 3;
+    const currentIndex = instruction.index;
+    // printout 3 previous instructions
+    if (currentIndex - previousInstructions + 1 > 0) {
```

Comment:
Style comment: I think this inequality is a little unclear.  I'd suggest rewriting it to `if (currentIndex - previousInstructions >= 0`.)  I also agre with Amal that it's worth adding a comment explaining what this inequality is doing.

Reply:
changes made